### ISSUE TITLE
Question about the pretrained weight

### ISSUE THREAD

**Issue by colin-de** [sahithchada/NOCS_PyTorch#5]
Hi! Thanks for the great work; I have some questions regarding the pre-trained weight provided in the repo. 
```
1. mask_rcnn_coco.pth      
2. resnet50_imagenet.pth
3. NOCS_Trained_1.pth      
4. NOCS_Trained_2.pth
```
what settings/experiments are those weights referring to? Thanks for your assistance!

**Comment by bharathsivaram10** [comment]
Hi Colin,

Happy to see you find our code useful. The weights are the following:

1 is the pre-trained weights from which you can start training the network. You can see in [train.py](https://github.com/sahithchada/NOCS_PyTorch/blob/f4ed85efec3b39476bda74ef84a314fcbcf8f0b3/train.py#L21)

2 is a set of weights for if you want to train just the mask rcnn on COCO data. I don't think we ever used this.

3 & 4 are trained sets of weights you can either use for NOCS testing since we have trained them, or you can load them and continue trying to train.

Hope this helps!

**Comment by colin-de** [comment]
Thanks for your information. If I'm using a dataset with different categories, is it recommended to train a model like `mask_rcnn_coco.pth `on new dataset first or can simply start with `rand_weight` option ?

using the pretrained weight from 1. (rand_weight=True) for training I got the inference result like this
![pred_coords](https://github.com/sahithchada/NOCS_PyTorch/assets/50893130/b1908165-adb2-4af2-8598-e46605513e1c)
much worse than the checkpoint from 4. NOCS_Trained_2.pth. 
![pred_coords](https://github.com/sahithchada/NOCS_PyTorch/assets/50893130/49cc3822-be0f-40ac-affc-1b0b7822fee1)
could you please let me know how to reproduce the training result. Many Thanks!

**Comment by sahithchada** [comment]
Hi Colin,

The inference results show that the mask head is not producing good segments in the image. Starting with pre-trained weights might ensure that the masks are well-formed.
To reproduce the results, you should start with the model pre-trained on COCO data.

**Comment by colin-de** [comment]
Thanks for info.
could you please let me know how the weight indices here are determined?
```
                weights = torch.reshape(weights, (81,4))
                w1 = weights[[0,40,46]]
                w2 = weights[[64,42]]
                w3 = torch.zeros_like(w2)
```
